src/loadConfig.js

```diff
@@ -0,0 +1,45 @@
+const path = require('path')
+const fs = require('fs')
+const YAML = require('yaml')
+
+const CUSTOM_FOLDER_PATH = 'custom/'
+const CONFIG_FILE_OVERRIDE_NAME = 'config'
+const SUPPORTED_EXTENSIONS = 'yaml|yml|json'
+const CONFIG_FILE = 'config-default.yaml'
+
+function parseJsonOrYaml(filePath) {
+  const extension = path.extname(filePath)
+  if (SUPPORTED_EXTENSIONS.split('|').includes(extension.replace('.', ''))) {
+    const content = fs.readFileSync(filePath, 'utf-8')
+    return YAML.parse(content)
+  } else {
+    throw new Error(`Unknown file extension "${extension}". Supported JSON or YAML: ${filePath} `)
+  }
+}
+
+function getCustomConfigFilePath() {
+  const customPath = path.resolve(CUSTOM_FOLDER_PATH)
+  const regex = new RegExp(`${CONFIG_FILE_OVERRIDE_NAME}\.(${SUPPORTED_EXTENSIONS})`)
+
+  const fileName = fs.readdirSync(customPath).find(fileName => regex.test(fileName))
```

Comment:
fine for me, although it looks simpler just to test the existance of the file for any of the extenssions, instead of going over all files in the dir, and testing for a pattern

sth in this direction:
```js
const extension = SUPPORTED_EXTENSIONS.find(extension => _existFile(CONFIG_FILE_OVERRIDE_NAME +  extension))
if (extension) {
   parseJsonOrYaml(CONFIG_FILE_OVERRIDE_NAME +  extension)
} else {
...
}
```